assets/js/modules/tagmanager/setup.js

```diff
@@ -406,12 +428,52 @@ class TagmanagerSetup extends Component {
 		);
 	}
 
-	render() {
+	/**
+	 * Render Error or Notice format depending on the errorCode.
+	 */
+	renderErrorOrNotice() {
 		const {
-			error,
-			message
+			errorCode,
+			errorMsg,
 		} = this.state;
 
+		const {
+			onSettingsPage,
+		} = this.props;
+
+		if ( ! errorCode ) {
+			return null;
+		}
+
+		let showError = true; // default error message.
+		let showNotice = false;
+		let message = errorMsg;
+
+		switch ( true ) {
+				case onSettingsPage && 'insufficientPermissions' === errorCode:
+					showError = false;
+					showNotice = true;
+					break;
+		}
+
+		if ( showError && 0 < message.length ) {
+			return (
+				<div className="googlesitekit-error-text">
+					<p>{ __( 'Error:', 'google-site-kit' ) } { message }</p>
+				</div>
+			);
+		}
+
+		if ( showNotice && 0 < message.length ) {
+			return (
+				<div>
+					<p>{ message }</p>
+				</div>
+			);
+		}
```

Comment:
It seems a bit unnecessary to render error and notice separately. You might consider combining these into a single JSX expression.